### ISSUE TITLE
Angular-cli + karma doesn't stop the test running even when compilation fails on Linux/MacOS machines

### ISSUE THREAD

**Issue by AlexanderSankin** [angular/angular-cli#6128]
### Bug Report or Feature Request (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
- [ ] feature request
```

### Versions.

Windows 10, MacOS Sierra 10.12.4, CentOS 7

kalupan@home:~/work/projects/test$ ng --version
@angular/cli: 1.0.1
node: 6.10.2
os: darwin x64
@angular/common: 4.1.0
@angular/compiler: 4.1.0
@angular/router: 4.1.0
@angular/core: 4.1.0
@angular/platform-browser: 4.1.0
@angular/platform-browser-dynamic: 4.1.0
@angular/cli: 1.0.1
@angular/compiler-cli: 4.1.0

kalupan@home:~/work/projects/test$ npm -v
4.5.0

### Repro steps.
Please download this small project
https://github.com/AlexanderSankin/test

See that class TestComponent has a deliberate compile error. It returns string instead of number type (https://github.com/AlexanderSankin/test/blob/master/src/test.component.ts)

Run npm install && ng test --no-watch. On Windows machine it will constantly fail the test as it should to be. On MacOS and Linux machine it will success. See logs below. 

### The log given by the failure.
**MacOS Sierra 10.12.4 logs (the same as on CentOS 7):**

```
kalupan@home:~/work/projects/test$ npm install && ng test --no-watch
npm WARN first@0.0.0 No description
npm WARN first@0.0.0 No repository field.
npm WARN first@0.0.0 No license field.
29 04 2017 13:39:29.640:DEBUG [config]: autoWatch set to false, because of singleRun
29 04 2017 13:39:29.643:DEBUG [plugin]: Loading inlined plugin (defining framework:jasmine).
29 04 2017 13:39:29.643:DEBUG [plugin]: Loading inlined plugin (defining launcher:Chrome, launcher:ChromeCanary, launcher:Chromium, launcher:Dartium, test).
29 04 2017 13:39:29.643:DEBUG [plugin]: Loading inlined plugin (defining launcher:PhantomJS).
29 04 2017 13:39:29.643:DEBUG [plugin]: Loading inlined plugin (defining framework:@angular/cli, preprocessor:@angular/cli, webpackPlugin, preprocessor:webpack, middleware:webpackBlocker, preprocessor:sourcemap).
29 04 2017 13:39:30.294:DEBUG [web-server]: Instantiating middleware
29 04 2017 13:39:36.949:DEBUG [preprocessor.sourcemap]: base64-encoded source map for /Users/kalupan/work/projects/test/src/polyfills.ts
29 04 2017 13:39:36.984:DEBUG [preprocessor.sourcemap]: base64-encoded source map for /Users/kalupan/work/projects/test/src/test.ts
29 04 2017 13:39:37.043:INFO [karma]: Karma v1.6.0 server started at http://0.0.0.0:9876/
29 04 2017 13:39:37.043:INFO [launcher]: Launching browser PhantomJS with unlimited concurrency
29 04 2017 13:39:37.072:INFO [launcher]: Starting browser PhantomJS
29 04 2017 13:39:37.072:DEBUG [temp-dir]: Creating temp dir at /var/folders/_r/d3x3wk2x4hb467v8l4f5yvc80000gn/T/karma-33714955
29 04 2017 13:39:37.075:DEBUG [launcher]: /Users/kalupan/work/projects/test/node_modules/phantomjs-prebuilt/lib/phantom/bin/phantomjs /var/folders/_r/d3x3wk2x4hb467v8l4f5yvc80000gn/T/karma-33714955/capture.js
29 04 2017 13:39:37.723:DEBUG [web-server]: serving: /Users/kalupan/work/projects/test/node_modules/karma/static/client.html
29 04 2017 13:39:37.729:DEBUG [web-server]: serving: /Users/kalupan/work/projects/test/node_modules/karma/static/karma.js
29 04 2017 13:39:37.761:DEBUG [karma]: A browser has connected on socket fFwbuz6JIQdwMIbkAAAA
29 04 2017 13:39:37.764:DEBUG [web-server]: upgrade /socket.io/?EIO=3&transport=websocket&sid=fFwbuz6JIQdwMIbkAAAA
29 04 2017 13:39:37.787:INFO [PhantomJS 2.1.1 (Mac OS X 0.0.0)]: Connected on socket fFwbuz6JIQdwMIbkAAAA with id 33714955
29 04 2017 13:39:37.788:DEBUG [launcher]: PhantomJS (id 33714955) captured in 0.744 secs
29 04 2017 13:39:37.790:DEBUG [phantomjs.launcher]: 
29 04 2017 13:39:37.791:DEBUG [middleware:karma]: custom files null null null
29 04 2017 13:39:37.791:DEBUG [middleware:karma]: Serving static request /context.html
29 04 2017 13:39:37.793:DEBUG [web-server]: serving: /Users/kalupan/work/projects/test/node_modules/karma/static/context.html
29 04 2017 13:39:37.795:DEBUG [middleware:source-files]: Requesting /base/src/polyfills.ts?db85e165c39e10352ae981ed0c76c0845f5c2653 /
29 04 2017 13:39:37.795:DEBUG [middleware:source-files]: Fetching /Users/kalupan/work/projects/test/src/polyfills.ts
29 04 2017 13:39:37.795:DEBUG [web-server]: serving (cached): /Users/kalupan/work/projects/test/src/polyfills.ts
29 04 2017 13:39:37.800:DEBUG [middleware:source-files]: Requesting /base/node_modules/karma-jasmine/lib/boot.js?945a38bf4e45ad2770eb94868231905a04a0bd3e /
29 04 2017 13:39:37.800:DEBUG [middleware:source-files]: Fetching /Users/kalupan/work/projects/test/node_modules/karma-jasmine/lib/boot.js
29 04 2017 13:39:37.800:DEBUG [middleware:source-files]: Requesting /base/node_modules/jasmine-core/lib/jasmine-core/jasmine.js?324c10193d2eee846e728831a0b54e7179ff5dc8 /
29 04 2017 13:39:37.800:DEBUG [middleware:source-files]: Fetching /Users/kalupan/work/projects/test/node_modules/jasmine-core/lib/jasmine-core/jasmine.js
29 04 2017 13:39:37.800:DEBUG [middleware:source-files]: Requesting /base/node_modules/karma-jasmine/lib/adapter.js?7a813cc290d592e664331c573a1a796192cdd1ad /
29 04 2017 13:39:37.800:DEBUG [middleware:source-files]: Fetching /Users/kalupan/work/projects/test/node_modules/karma-jasmine/lib/adapter.js
29 04 2017 13:39:37.800:DEBUG [middleware:source-files]: Requesting /base/src/test.ts?8c2d05565a259342f40be89b424e0a0ca82e0fa0 /
29 04 2017 13:39:37.800:DEBUG [middleware:source-files]: Fetching /Users/kalupan/work/projects/test/src/test.ts
29 04 2017 13:39:37.801:DEBUG [web-server]: serving (cached): /Users/kalupan/work/projects/test/node_modules/karma-jasmine/lib/boot.js
29 04 2017 13:39:37.801:DEBUG [web-server]: serving (cached): /Users/kalupan/work/projects/test/node_modules/jasmine-core/lib/jasmine-core/jasmine.js
29 04 2017 13:39:37.801:DEBUG [web-server]: serving (cached): /Users/kalupan/work/projects/test/node_modules/karma-jasmine/lib/adapter.js
29 04 2017 13:39:37.802:DEBUG [web-server]: serving (cached): /Users/kalupan/work/projects/test/src/test.ts
29 04 2017 13:39:37.834:DEBUG [web-server]: serving: /Users/kalupan/work/projects/test/node_modules/karma/static/context.js
PhantomJS 2.1.1 (Mac OS X 0.0.0): Executed 1 of 1 SUCCESS (0.025 secs / 0.077 secs)
29 04 2017 13:39:38.065:DEBUG [karma]: Run complete, exiting.
29 04 2017 13:39:38.065:DEBUG [launcher]: Disconnecting all browsers
29 04 2017 13:39:38.078:DEBUG [launcher]: Process PhantomJS exited with code 0
29 04 2017 13:39:38.078:DEBUG [temp-dir]: Cleaning temp dir /var/folders/_r/d3x3wk2x4hb467v8l4f5yvc80000gn/T/karma-33714955
29 04 2017 13:39:38.081:DEBUG [launcher]: Finished all browsers
```

**Windows 10 logs:**

```
Admin@DESKTOP-1VN66LF MINGW64 /c/first
$ ng test --no-watch
29 04 2017 03:40:43.300:DEBUG [config]: autoWatch set to false, because of singleRun
29 04 2017 03:40:43.300:DEBUG [plugin]: Loading inlined plugin (defining framework:jasmine).
29 04 2017 03:40:43.300:DEBUG [plugin]: Loading inlined plugin (defining launcher:Chrome, launcher:ChromeCanary, launcher:Chromium, launcher:Dartium, test).
29 04 2017 03:40:43.300:DEBUG [plugin]: Loading inlined plugin (defining launcher:PhantomJS).
29 04 2017 03:40:43.300:DEBUG [plugin]: Loading inlined plugin (defining framework:@angular/cli, preprocessor:@angular/cli, webpackPlugin, preprocessor:webpack, middleware:webpackBlocker, preprocessor:sourcemap).
29 04 2017 03:40:44.160:DEBUG [web-server]: Instantiating middleware
29 04 2017 03:40:52.285:ERROR [karma]: Error: C:/first/src/test.component.ts (10,5): Type '"555"' is not assignable to type 'number'.
    at Compiler.compiler.plugin (C:\first\node_modules\@angular\cli\plugins\karma-webpack-throw-error.js:10:23)
    at Compiler.applyPlugins (C:\first\node_modules\tapable\lib\Tapable.js:61:14)
    at Watching._done (C:\first\node_modules\webpack\lib\Compiler.js:91:17)
    at onCompiled (C:\first\node_modules\webpack\lib\Compiler.js:50:17)
    at C:\first\node_modules\webpack\lib\Compiler.js:480:13
    at next (C:\first\node_modules\tapable\lib\Tapable.js:138:11)
    at Compiler.<anonymous> (C:\first\node_modules\webpack\lib\CachePlugin.js:62:5)
    at Compiler.applyPluginsAsyncSeries (C:\first\node_modules\tapable\lib\Tapable.js:142:13)
    at C:\first\node_modules\webpack\lib\Compiler.js:477:10
    at Compilation.applyPluginsAsyncSeries (C:\first\node_modules\tapable\lib\Tapable.js:131:46)
    at self.applyPluginsAsync.err (C:\first\node_modules\webpack\lib\Compilation.js:640:19)
    at next (C:\first\node_modules\tapable\lib\Tapable.js:138:11)
    at Compilation.<anonymous> (C:\first\node_modules\webpack\lib\ProgressPlugin.js:122:5)
    at Compilation.applyPluginsAsyncSeries (C:\first\node_modules\tapable\lib\Tapable.js:142:13)
    at self.applyPluginsAsync.err (C:\first\node_modules\webpack\lib\Compilation.js:631:11)
    at next (C:\first\node_modules\tapable\lib\Tapable.js:138:11)
29 04 2017 03:40:52.285:DEBUG [launcher]: Disconnecting all browsers
```

### Desired functionality.

It should always stop the test run because of compilation error.

### Mention any other details that might be useful.

Please note - if you make some changes in TestComponent (for example change `return "555"` to `return "777"`) it will fail the test only at first run (see log below). But all subsequent runs will be successful again!

**First run ng test --no-watch after some changes**

```
kalupan@home:~/work/projects/test$ ng test --no-watch
29 04 2017 13:55:26.720:DEBUG [config]: autoWatch set to false, because of singleRun
29 04 2017 13:55:26.724:DEBUG [plugin]: Loading inlined plugin (defining framework:jasmine).
29 04 2017 13:55:26.724:DEBUG [plugin]: Loading inlined plugin (defining launcher:Chrome, launcher:ChromeCanary, launcher:Chromium, launcher:Dartium, test).
29 04 2017 13:55:26.724:DEBUG [plugin]: Loading inlined plugin (defining launcher:PhantomJS).
29 04 2017 13:55:26.724:DEBUG [plugin]: Loading inlined plugin (defining framework:@angular/cli, preprocessor:@angular/cli, webpackPlugin, preprocessor:webpack, middleware:webpackBlocker, preprocessor:sourcemap).
29 04 2017 13:55:27.377:DEBUG [web-server]: Instantiating middleware
29 04 2017 13:55:34.936:DEBUG [preprocessor.sourcemap]: base64-encoded source map for /Users/kalupan/work/projects/test/src/polyfills.ts
29 04 2017 13:55:34.965:DEBUG [preprocessor.sourcemap]: base64-encoded source map for /Users/kalupan/work/projects/test/src/test.ts
29 04 2017 13:55:35.018:INFO [karma]: Karma v1.6.0 server started at http://0.0.0.0:9876/
29 04 2017 13:55:35.018:INFO [launcher]: Launching browser PhantomJS with unlimited concurrency
29 04 2017 13:55:35.022:INFO [launcher]: Starting browser PhantomJS
29 04 2017 13:55:35.022:DEBUG [temp-dir]: Creating temp dir at /var/folders/_r/d3x3wk2x4hb467v8l4f5yvc80000gn/T/karma-25384356
29 04 2017 13:55:35.025:DEBUG [launcher]: /Users/kalupan/work/projects/test/node_modules/phantomjs-prebuilt/lib/phantom/bin/phantomjs /var/folders/_r/d3x3wk2x4hb467v8l4f5yvc80000gn/T/karma-25384356/capture.js
29 04 2017 13:55:35.376:ERROR [karma]: Error: /Users/kalupan/work/projects/test/src/test.component.ts (10,5): Type '"777"' is not assignable to type 'number'.
    at Compiler.compiler.plugin (/Users/kalupan/work/projects/test/node_modules/@angular/cli/plugins/karma-webpack-throw-error.js:10:23)
    at Compiler.applyPlugins (/Users/kalupan/work/projects/test/node_modules/tapable/lib/Tapable.js:61:14)
    at Watching._done (/Users/kalupan/work/projects/test/node_modules/webpack/lib/Compiler.js:91:17)
    at onCompiled (/Users/kalupan/work/projects/test/node_modules/webpack/lib/Compiler.js:50:17)
    at /Users/kalupan/work/projects/test/node_modules/webpack/lib/Compiler.js:480:13
    at next (/Users/kalupan/work/projects/test/node_modules/tapable/lib/Tapable.js:138:11)
    at Compiler.<anonymous> (/Users/kalupan/work/projects/test/node_modules/webpack/lib/CachePlugin.js:62:5)
    at Compiler.applyPluginsAsyncSeries (/Users/kalupan/work/projects/test/node_modules/tapable/lib/Tapable.js:142:13)
    at /Users/kalupan/work/projects/test/node_modules/webpack/lib/Compiler.js:477:10
    at Compilation.applyPluginsAsyncSeries (/Users/kalupan/work/projects/test/node_modules/tapable/lib/Tapable.js:131:46)
    at self.applyPluginsAsync.err (/Users/kalupan/work/projects/test/node_modules/webpack/lib/Compilation.js:640:19)
    at next (/Users/kalupan/work/projects/test/node_modules/tapable/lib/Tapable.js:138:11)
    at Compilation.<anonymous> (/Users/kalupan/work/projects/test/node_modules/webpack/lib/ProgressPlugin.js:122:5)
    at Compilation.applyPluginsAsyncSeries (/Users/kalupan/work/projects/test/node_modules/tapable/lib/Tapable.js:142:13)
    at self.applyPluginsAsync.err (/Users/kalupan/work/projects/test/node_modules/webpack/lib/Compilation.js:631:11)
    at next (/Users/kalupan/work/projects/test/node_modules/tapable/lib/Tapable.js:138:11)
29 04 2017 13:55:35.391:DEBUG [launcher]: Disconnecting all browsers
29 04 2017 13:55:35.397:DEBUG [launcher]: Process PhantomJS exited with code 0
29 04 2017 13:55:35.398:DEBUG [temp-dir]: Cleaning temp dir /var/folders/_r/d3x3wk2x4hb467v8l4f5yvc80000gn/T/karma-25384356
29 04 2017 13:55:35.401:DEBUG [launcher]: Finished all browsers
```

The same behavior with Chrome browser **ng test --no-watch --browsers='Chrome'**

**Comment by sumitarora** [comment]
@AlexanderSankin This has been fixed in `1.1.0-beta.1` release for Angular CLI with performance improvements https://github.com/angular/angular-cli/commit/1cd0a08

**Comment by filipesilva** [comment]
Closing then, thanks @sumitarora!
